pkg/mutation/assign_mutator.go

```diff
@@ -91,6 +135,9 @@ func (m *AssignMutator) DeepCopy() types.Mutator {
 	}
 	copy(res.path.Nodes, m.path.Nodes)
 	copy(res.bindings, m.bindings)
+	m.mux.RLock()
```

Comment:
Sure, the reason is that the others are created on start-up via MutatorForAssign.

I could move the initialization there instead of lazy-loading to avoid locking entirely?

Reply:
That would have the added benefit of avoiding the need to have as many `error` returns as we do currently.

Reply:
Agreed - I feel like it simplifies things if we only need to deal with a fully initialized mutator. 👍🏻

Reply:
addressed